### PR TITLE
JS config files 

### PR DESCRIPTION
**Pull request by jhliberty**
changes config files to be js instead of json files so env variables and such can be used.

Also updates the example apps.

### CODE REVIEW

**Comment by jhliberty** [comment]
should probably npm link in ci so it uses the current version that's being tested

File: .travis.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ services:
 before_script:
   - cat test/fixtures/data.sql | mysql -u travis
   - npm install -g gulp
-  - npm install -g lux-framework
+  - npm link
```
Comment (zacharygolba) [inline .travis.yml]: This is definitely a must. Thanks man! Can you do this in a separate PR? It would be nice to have record of this change after we squash and merge this PR.

File: examples/social-network/config/database.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  username: 'root',
+  password: 'root',
+  database: 'social_network_prod',
+  host: '127.0.0.1',
+  port: '3306',
+  dialect: 'mysql',
+  logging: false
+};
+
+export { development, test, production };
```
Comment (zacharygolba) [inline examples/social-network/config/database.js]: This looks great. One suggestion though.

For consistency with the previous structure of the json config file, we should `export default` and object literal of the config.

```javascript
export default {
  development: {
    /* ... */
  },

  test: {
    /* ... */
  },

  production: {
    /* ... */
  }
};
```

File: src/packages/cli/commands/serve.js [hunk 3]
```diff
@@ -10,7 +10,7 @@ const env = process.env.NODE_ENV || 'development';
 
 export default async function serve(port = 4000) {
   const Application = require(`${pwd}/bin/app`);
-  const config = require(`${pwd}/config/environments/${env}.json`);
+  const config = require(`${pwd}/config/environments/${env}.js`);
```
Comment (zacharygolba) [inline src/packages/cli/commands/serve.js]: We'll have to change this to:

```javascript
const { default: config } = require(`${pwd}/config/environments/${env}`);
```

File: src/packages/application/index.js [hunk 4]
```diff
@@ -30,7 +30,7 @@ class Application extends Base {
   async boot() {
     const { root, router, domain, server, port } = this;
     const store = Database.create(
-      require(`${root}/config/database.json`)
+      require(`${root}/config/database.js`)
```
Comment (zacharygolba) [inline src/packages/application/index.js]: We can drop the `.js` from require. Also after refactoring to `export default` from the config files we'll have to update this to:

```javascript
const store = Database.create(
  require(`${root}/config/database`).default
);
```

**Comment by zacharygolba** [comment]
This is awesome man! We definitely needed this. This should be good to merge after the few minor updates I suggested above.Thanks again for your help. 😃

File: test/unit/packages/session.js [hunk 5]
```diff
@@ -3,7 +3,7 @@ import { randomBytes } from 'crypto';
 import fetch from 'isomorphic-fetch';
 
 import encrypt from '../../../src/packages/session/utils/encrypt';
-import config from '../../test-app/config/environments/test.json';
+var { default: config } = require('../../test-app/config/environments/test');
```
Comment (zacharygolba) [inline test/unit/packages/session.js]: One change to this line and this PR should be good to squash and merge 👍.

```javascript
import config from '../../test-app/config/environments/test';
```

**Comment by zacharygolba** [comment]
🎊 Great job @jhliberty!
